cheatsheets/Microservices_Security_Cheat_Sheet.md

```diff
@@ -2,176 +2,660 @@
+
+Each microservice continues to act as the PEP, calling the central PDP during request handling to obtain access decisions. The PDP requires relevant attributes to evaluate access. In PBAC systems, these attributes may either be passed in by the service or fetched by the PDP itself, depending on its capabilities. ReBAC and NGAC systems usually expect most relevant attributes and relationships to be stored in their internal databases, though some (such as SpiceDB, or OpenFGA already referenced above) allow limited attribute injection at request time.
+
+Although this pattern improves observability and supports a broader range of access control models, it introduces its own trade-offs and does not eliminate all challenges found in the previous pattern.
+
+#### Pros
+
+* **Policy governance:** Policies can be centrally defined, versioned, reviewed, and audited, independent of the service’s implementation language.
+* **Policy layering:** The model allows for both global (e.g. security team–defined) and local (e.g. service team–defined) policies to coexist. This enables clearer separation of concerns and better alignment with organizational structure and responsibilities.
+* **Improved monitoring:** All decisions can be consistently logged and monitored, assuming proper instrumentation.
+* **Team autonomy:** Teams remain responsible for their services and their integration with the PDP. This aligns well with independent team ownership and domain-driven design principles.
+* **Support for "before-the-fact" audit:** Particularly with ReBAC and NGAC systems as PDP, authorization models allow querying the existing access rights making answering the corresponding questions a simple game.
+* **Support for additional PDP models:** This pattern enables the use of a broader set of access control models, including ReBAC and NGAC systems, which may better fit for a given business context.
+
+#### Cons
+
+* **Policy distribution complexity:** Policies are managed independently of service code, and the PDP may support multiple services and domains. Ensuring the correct policy version is applied consistently can be complex.
+* **Context sharing:** PDPs do not inherently have access to service context. Mechanisms must be designed to assemble and pass the right attributes into the PDP for evaluation and in case of the ReBAC/NGAC systems to build the actual authorization model.
+* **Performance overhead:** Network hops between the microservice and the external PDP introduce latency and a potential point of contention under high load.
```

Comment:
just a digression, nothing to change: at least spiceDB is also usable as a library -> https://github.com/authzed/examples/tree/main/spicedb-as-library - but it's "just" a community example, and iirc experimental, so not really worth mentioning i guess, also bc imo rebac systems (or at least spiceDB) are really "designed" to work as their own, highly available and horizontally distributable service.

It might come in handy when you want to build your own abstraction around SpiceDB, though, or enhance it with a wrapper for more fine-grained meta-access-control currently only available in the paid version, etc... but as said: I digress. Nevermind ;)